src/main/java/org/opensearch/geospatial/ip2geo/jobscheduler/DatasourceRunner.java

```diff
@@ -118,161 +106,50 @@ public void runJob(final ScheduledJobParameter jobParameter, final JobExecutionC
-                        .setIndicesOptions(IndicesOptions.LENIENT_EXPAND_OPEN)
-                        .execute()
-                        .actionGet(timeout)
-                        .isAcknowledged()) {
-                        deletedIndices.add(index);
-                    } else {
-                        log.error("Failed to delete an index [{}]", index);
-                    }
-                } catch (Exception e) {
-                    log.error("Failed to delete an index [{}]", index, e);
+                    updateDatasource(jobParameter);
+                } finally {
+                    lockService.release(
+                        lock,
+                        ActionListener.wrap(released -> {}, exception -> { log.error("Failed to release lock [{}]", lock, exception); })
+                    );
                 }
-            }
-            if (!deletedIndices.isEmpty()) {
-                parameter.getIndices().removeAll(deletedIndices);
-                DatasourceHelper.updateDatasource(client, parameter, timeout);
-            }
-        } catch (Exception e) {
-            log.error("Failed to delete old indices for {}", parameter.getId(), e);
-        }
+            }, exception -> { log.error("Failed to acquire lock for job [{}]", jobParameter.getName(), exception); }));
+        };
     }
 
-    /**
-     * Update GeoIP data internal
-     *
-     * @param jobParameter
-     * @throws Exception
-     */
-    private void updateDatasource(final Datasource jobParameter) throws Exception {
-        if (!DatasourceState.AVAILABLE.equals(jobParameter.getState())) {
-            log.error("Invalid datasource state. Expecting {} but received {}", DatasourceState.AVAILABLE, jobParameter.getState());
-            jobParameter.disable();
-            jobParameter.getUpdateStats().setLastFailedAt(Instant.now());
-            DatasourceHelper.updateDatasource(client, jobParameter, timeout);
+    @VisibleForTesting
+    protected void updateDatasource(final ScheduledJobParameter jobParameter) throws IOException {
+        Datasource datasource = datasourceFacade.getDatasource(jobParameter.getName());
+        if (datasource == null) {
+            log.info("Datasource[{}] does not exist", jobParameter.getName());
             return;
         }
 
-        URL url = new URL(jobParameter.getEndpoint());
-        DatasourceManifest manifest = DatasourceManifest.Builder.build(url);
-
-        if (skipUpdate(jobParameter, manifest)) {
-            log.info("Skipping GeoIP database update. Update is not required for {}", jobParameter.getId());
-            jobParameter.getUpdateStats().setLastSkippedAt(Instant.now());
-            DatasourceHelper.updateDatasource(client, jobParameter, timeout);
+        if (!DatasourceState.AVAILABLE.equals(datasource.getState())) {
+            log.error("Invalid datasource state. Expecting {} but received {}", DatasourceState.AVAILABLE, datasource.getState());
+            datasource.disable();
```

Comment:
Why is this being disabled?

Reply:
Otherwise, the next update will be scheduled but we don't want that to happen as it will fail again anyway.